core/directory.go

```diff
@@ -257,15 +257,42 @@ func (dir *Directory) Entries(ctx context.Context, bk *buildkit.Client, svcs *Se
 	}
 
 	entries, err := ref.ReadDir(ctx, bkgw.ReadDirRequest{
-		Path: src,
+		Path:           src,
+		IncludePattern: include,
 	})
 	if err != nil {
 		return nil, err
 	}
 
+	// Check if `include` contains a recursive glob pattern
+	recursive := strings.Contains(include, "**")
```

Comment:
I'm not sure about this - it would be nice to avoid introspecting this string. But I think it works for now. If we had multiple include patterns, this would the logic for this would be quite trick I think - one pattern could be recursive, while the other wouldn't be.

Wdyt about instead *always* enabling `recursive = true` if `include` isn't empty? I don't see a use case where `include` is set without a recursive path and `src` is also set.

This is sort of what I meant about having a separate `Glob` method - it's really annoying to work out when the recursive case should occur. I don't want to block, but still think that might be the way to go.

Reply:
Also the different path manipulation based on `recursive` is quite confusing. I think there's such differences in the return values (they're really not interchangeable), that I think that adds more weight to splitting it out to another method.

Reply:
If you do `include: *.md`, you only expect files that match the pattern in the current directory

Reply:
If `include: *.md`, then `ReadDir` only returns files in the current directory, and never anything below.

Reply:
If I do that, the issue is that output will include the `src` if it's set where we only expect it to be print for sub directories